Jingyi Lin- Assignment 2.Rmd

```diff
@@ -0,0 +1,134 @@
+names(V.TCH) <- c("id", "gender", "topic", "group")
+names(V.STD) <- c("id", "gender", "topic", "group")
+
+#Bind the two data frames together (you will get a warning because the teacher data frame has 5 types of id (A,B,C,D,E) and the student has 25 (1-30), this isn't a problem)
+VERTEX <- dplyr::bind_rows(V.TCH, V.STD)
+
+```
+
+Now we have both a Vertex and Edge list it is time to plot our graph!
+
+```{r}
+#Load the igraph package
+
+library(igraph)
+
+#First we will make an object that contains the graph information using our two dataframes EDGE and VERTEX. Notice that we have made "directed = TRUE" - our graph is directed since discipline is being given from a teacher to a student.
+
+g <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX)
+
+#Now we can plot our graph using the force directed graphing technique - our old friend Fruchertman-Reingold!
+
+plot(g,layout=layout.fruchterman.reingold)
+
+#There are many ways to change the attributes of the graph to represent different characteristics of the newtork. For example, we can color the nodes according to gender.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+
+#We can change the thickness of the edge according to the number of times a particular teacher has sent a particular student to the principal.
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+````
+
+## Part II
+
+In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+
+```{r}
+N.tid <- EDGE %>% group_by(from) %>% summarise(sum(count))
+N.sid <- EDGE %>% group_by(to) %>% summarise(sum(count))
+names(N.tid) <- c("id", "count")
+names(N.sid) <- c("id", "count")
+N <-dplyr::bind_rows(N.tid, N.sid)
+g1 <- graph.data.frame(N, directed=TRUE, vertices=VERTEX)
+plot(g1,layout=layout.fruchterman.reingold,vertex.color=VERTEX$gender)
+```
+
+
+## Part III
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+```{r}
+H1 <- read.csv(file = "HUDK4050-2018-classes.csv", header = TRUE)
+library(tidyr)
+H2 <- select(H1,"Q8","Q9","Q1","Q3","Q4","Q5","Q6")
+H3 <- H2[-c(1,2), ]
+H4 <- H3 %>% unite(First_Last, Q8, Q9, sep="_", remove=TRUE) %>% print
+which(duplicated(H4$First_Last))
+H4_no_duplicate <- H4[-c(1), ]
+H5 <- gather(H4_no_duplicate,course.label, course, Q1,Q3,Q4,Q5,Q6, na.rm = TRUE, convert = FALSE) 
+H5 <- select(H5, First_Last, course)
+H5 <- filter(H5, course>0,course != "HUDK 4050")
+H5$Count <- 1
+H6 <- spread(H5, course, Count)
```

Comment:
Well done! I difficult problem that you have solved effectively and efficiently!